BridgeAppSDK/SBAStepHeaderView.swift

```diff
@@ -0,0 +1,380 @@
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import UIKit
+
+/**
+ A custom UIView to be included in an SBAGenericStepViewController. It optionally contains several subViews
+ and displays them in this order, from top to bottom of the view:
+ 
+ 1) progressView: SBAStepProgressView - show progress thru the current flow
+ 2) imageView: UIImageView - shows an image associated with the current step
+ 3) headerLabel: UILabel - generally the Title of the current step
+ 4) detailsLabel: UILabel - generally the Text of the current step
+ 5) customView: UIView - any custom view provided by the SBAGenericStepViewController
+ 6) learnMoreButton: UIButton - a button to call the learnMoreAction
+ 7) promptLabel: UILabel - a label intended to prompt the user to enter data or make a selection
+ 
+ Several public properties are provided to configure the view, such has hiding or showing the learnMoreButton
+ or progressView, and providing a minimumHeight or customView.
+ 
+ To customize the view elements, subclasses should override the initializeViews() method. This will allow
+ the use of any custom element (of the appropriate type) to be used instead of the default instances.
+ */
+
+open class SBAStepHeaderView: UIView {
+    
+    private let kTopMargin: CGFloat = CGFloat(30.0).proportionalToScreenWidth()
+    private let kSideMargin: CGFloat = CGFloat(30.0).proportionalToScreenWidth()
+    private let kVerticalSpacing: CGFloat = CGFloat(20.0).proportionalToScreenWidth()
+    private let kBottomMargin: CGFloat = CGFloat(30.0).proportionalToScreenWidth()
+    private let kPromptBottomMargin: CGFloat = CGFloat(10.0).proportionalToScreenWidth()
+    private let kImageViewHeight: CGFloat = CGFloat(100.0).proportionalToScreenWidth()
+    private let kLearnMoreButtonHeight: CGFloat = 30.0
+    private let kLabelMaxLayoutWidth: CGFloat = {
+        return CGFloat(UIScreen.main.bounds.size.width - (2 * CGFloat(30.0).proportionalToScreenWidth()))
+    }()
```

Comment:
Hmmm... not sure. It's not a huge deal for our current apps (which are constrained to portrait), but something to keep in mind for iPad, etc.